### PR TITLE
Create new deal details

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
*Replace this paragraph with a description of what this PR is changing or adding, and why.*

*List which issues are fixed by this PR. You must list at least one issue.*

## Checklist

- [ ] I used the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
- [ ] Unit tests have been created if a new feature was added.
- [ ] Translations have been added and extracted if a new feature was added.

### CODE REVIEW

File: src/components/AmortizationRepaymentSchedule.tsx [hunk 1]
```diff
@@ -1,125 +1,83 @@
...
+import { Button } from "@components/Button";
+import { repaymentSchedule } from "@utils/amortization.utils";
+import { generateGraphAndTableData } from "@utils/repayment.utils";
+import { TableProps } from "@components/Table";
+import { Ratio } from "@credix/credix-client";
 
 interface AmortizationRepaymentScheduleProps {
+	principal: number;
+	financingFee: number;
+	repaymentPeriod: number;
```
Comment (RuvendeGroote) [inline src/components/AmortizationRepaymentSchedule.tsx]: should be plural

File: src/components/AmortizationRepaymentSchedule.tsx [hunk 2]
```diff
@@ -1,125 +1,83 @@
...
-	for (let i = 1; i <= 12; i++) {
-		data.push({
-			month: i.toString(),
-			type: RepaymentScheduleAmountType.INTEREST,
-			amount: 120 - i * 10,
-		});
+	useEffect(() => {
+		if (principal && financingFee && repaymentPeriod) {
+			const financingFeeRatio = new Ratio(financingFee, 100);
+			const schedule = repaymentSchedule(principal, financingFeeRatio, repaymentPeriod);
```
Comment (RuvendeGroote) [inline src/components/AmortizationRepaymentSchedule.tsx]: I don't really like that this has the same name for bullet loans and amortization loans. it's not really apparent that we are using the wrong one unless you check the imports (which most people don't). anyways, not a big deal ig as the schedule generation is moving to the client

**Comment by RuvendeGroote** [comment]
image
I have noticed some visual regression. not sure if it's because of this PR

File: src/utils/format.utils.ts [hunk 3]
```diff
@@ -14,6 +14,11 @@ export const numberFormatter = Intl.NumberFormat("en", {
 	maximumFractionDigits: 1,
 });
 
+export const currencyFormatter = Intl.NumberFormat("en", {
```
Comment (RuvendeGroote) [inline src/utils/format.utils.ts:17]: what is your opinion on renaming the numberFormatter to something like compactformatter? for me it was confusing why we had two seemingly similar functions (made sense when I saw how they were used though)

File: src/components/SelectorCard.tsx [hunk 4]
```diff
@@ -10,6 +10,9 @@ interface SelectorCardProps {
 	value: string;
 	checked?: boolean;
 	showContent?: boolean;
+	isDisplay?: boolean;
+	className?: string;
+	selectCard?: () => void;
```
Comment (RuvendeGroote) [inline src/components/SelectorCard.tsx]: I prefer 'on{action}' names for click/event handlers

File: src/components/SelectorCard.tsx [hunk 5]
```diff
@@ -20,21 +23,33 @@ export const SelectorCard: FunctionComponent<SelectorCardProps> = ({
...
+	const classes = classNames([
+		className,
+		"p-6 border",
+		isDisplay ? "border-neutral-40" : "border-dashed rounded-md border-black hover:cursor-pointer",
 		checked && "bg-white",
 	]);
 
 	return (
-		<div className={className}>
+		<div className={classes} onClick={selectCard}>
```
Comment (RuvendeGroote) [inline src/components/SelectorCard.tsx]: we have an onClick on the outer div that will still trigger even if it's `isDisplay`. is this behaviour we want? I can imagine if we say it is display only that we don't want to make it interactable

File: src/components/SelectorCard.tsx [hunk 6]
```diff
@@ -20,21 +23,33 @@ export const SelectorCard: FunctionComponent<SelectorCardProps> = ({
 	checked,
 	value,
 	showContent,
+	isDisplay,
```
Comment (RuvendeGroote) [inline src/components/SelectorCard.tsx]: I prefer displayOnly or interactive or something

File: src/components/RepaymentScheduleTable.tsx [hunk 7]
```diff
@@ -16,18 +20,71 @@ export interface RepaymentScheduleDataPoint {
...
 export const RepaymentScheduleTable: FunctionComponent<RepaymentScheduleTableProps> = ({
 	dataSource,
-	columns,
 }) => {
+	const intl = useIntl();
+	const locales = useLocales();
+
+	const columns: ColumnsProps[] = [
+		{
+			title: "Date",
```
Comment (RuvendeGroote) [inline src/components/RepaymentScheduleTable.tsx]: use intl

File: src/components/RepaymentScheduleTable.tsx [hunk 8]
```diff
@@ -16,18 +20,71 @@ export interface RepaymentScheduleDataPoint {
 
 interface RepaymentScheduleTableProps {
 	dataSource: AntdTableProps<any>["dataSource"];
```
Comment (RuvendeGroote) [inline src/components/RepaymentScheduleTable.tsx]: we don't have to use any here as we know the dataSource type

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

**Comment by BartDeCaluwe** [comment]
> image
> I have noticed some visual regression. not sure if it's because of this PR

fixed in: https://github.com/credix-finance/credix-app-v2/pull/148

File: src/components/AmortizationRepaymentSchedule.tsx (hunk 2 above)
Comment (BartDeCaluwe) [reply src/components/AmortizationRepaymentSchedule.tsx]: Yes totally agree. I wasn't too happy about it myself.

File: src/utils/format.utils.ts (hunk 3 above)
Comment (BartDeCaluwe) [reply src/utils/format.utils.ts:17]: We can def do that. `numberFormatter` is very generic for what it actually does.

File: src/utils/repayment.utils.ts [hunk 9]
```diff
@@ -0,0 +1,34 @@
+import { RepaymentScheduleAmountType } from "@components/RepaymentScheduleGraph";
+import { Repayment } from "@utils/amortization.utils";
+
+export const generateGraphAndTableData = (schedule: Repayment[]) => {
+	const currentDate = new Date();
+
+	return schedule.reduce(
+		(acc, repayment, index) => {
```
Comment (RuvenSalamon) [inline src/utils/repayment.utils.ts:8]: can we put these in types?

Comment (RuvendeGroote) [inline src/utils/repayment.utils.ts:12]: can we put these in types?

File: src/components/DealForm.tsx [hunk 10]
```diff
@@ -24,169 +23,51 @@ interface DealFormProps {
...
 
-	const validateBorrowerKey = (value): Promise<void> => {
-		const validationMessage = intl.formatMessage({
-			defaultMessage: "'borrower key' isn't valid",
-			description: "Deal form: borrower key validation message",
-		});
-		return validatePublicKey(value, validationMessage);
+	const onNextStep = async (fieldsToValidate: string[], nextStep: number) => {
+		try {
+			await form.validateFields(fieldsToValidate);
```
Comment (RuvendeGroote) [inline src/components/DealForm.tsx]: `await form.validateFields(fieldsToValidate).then(() => setCurrentStep(nextStep))`

Comment (RuvendeGroote) [reply src/components/DealForm.tsx]: no try empty catch needed

File: src/components/AmortizationRepaymentSchedule.tsx [hunk 11]
```diff
@@ -1,125 +1,83 @@
...
+							description: "Deal form: repayment schedule grace period input description",
+						})}
+						placeholder={intl.formatMessage({
+							defaultMessage: "0 days",
+							description: "Deal form: repayment schedule grace period input placeholder",
+						})}
 						disabled={true}
 					/>
 					<div>
 						<Button onClick={() => setShowTable(!showTable)} type="text">
```
Comment (RuvendeGroote) [inline src/components/AmortizationRepaymentSchedule.tsx]: this is duplicate in both schedules

File: src/components/AmortizationRepaymentSchedule.tsx [hunk 12]
```diff
@@ -1,125 +1,83 @@
...
-			key: "balance",
-		},
-	];
+			const { graphData, dataSource } = generateGraphAndTableData(schedule);
+			setGraphData(graphData);
+			setDataSource(dataSource);
+		}
+	}, [principal, financingFee, timeToMaturity]);
 
 	return (
```
Comment (RuvendeGroote) [inline src/components/AmortizationRepaymentSchedule.tsx]: this component is duplicate except for the input and the logic

File: src/utils/amortization.utils.ts [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+import { Ratio } from "@credix/credix-client";
+import Big from "big.js";
+import { round } from "./format.utils";
+
+export const calculateMonthlyPayment = (
+	principal: number,
+	rate: Ratio,
+	numberOfPayments: number
+) => {
+	if (!rate) {
```
Comment (RuvendeGroote) [inline src/utils/amortization.utils.ts]: rate is always present according to the function.

File: src/utils/amortization.utils.ts [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+		rate = new Ratio(1, 1);
+	}
+
+	if (!numberOfPayments) {
```
Comment (RuvendeGroote) [inline src/utils/amortization.utils.ts]: same here

File: src/consts.ts [hunk 15]
```diff
@@ -33,3 +36,69 @@ export const dealsRoute: Route = {
 };
 
 export const routes: Route[] = [investWithdrawRoute, dealsRoute];
+
+export const defaultTranches = [
```
Comment (RuvendeGroote) [inline src/consts.ts:41]: this seems unused

File: package.json [hunk 16]
```diff
@@ -23,7 +24,7 @@
 	"dependencies": {
 		"@ant-design/charts": "^1.3.6",
 		"@civic/solana-gateway-react": "0.4.13",
-		"@credix/credix-client": "^0.1.0-beta.4",
+		"@credix/credix-client": "^0.1.0",
```
Comment (RuvendeGroote) [inline package.json]: is this the correct client?

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

**Review by RuvendeGroote — APPROVED** [review]

**Pull request merged**